pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -434,6 +435,16 @@ func getNICIPAddresses(ctx context.Context, c *config, ifaceName string) (map[st
 				for _, ip := range publicIPs {
 					ipAddresses[ip] = v1.NodeExternalIP
 				}
+
+				publicIP6Name := ifaceName + "-pubipv6"
```

Comment:
nit: may be a good idea to use constants for these interface name suffixes (`-pubip`, `-pubipv6`)